java-client/src/test/java/org/opensearch/client/opensearch/integTest/AbstractCatClientIT.java

```diff
@@ -0,0 +1,206 @@
+
+        assertTrue(aliasRecord.index().equals(indexName));
+        assertTrue(aliasRecord.isWriteIndex().equals(Boolean.TRUE.toString()));
+        assertNotNull(aliasRecord.routingIndex());
+        assertNotNull(aliasRecord.routingSearch());
+        assertNull(aliasRecord.filter());
+    }
+
+    public void testCatShards() throws Exception {
+        createIndex("test-cat-shards-index-1");
+        createIndex("test-cat-shards-index-3");
+        createIndex("test-cat-shards-index-2");
+        createIndex("a-test-cat-shards-index");
+
+        ShardsResponse shardsResponse = javaClient().cat().shards(r -> r.headers("index,shard,prirep,state,store").sort("index")
+                .bytes(Bytes.Bytes));
+        assertNotNull(shardsResponse.valueBody());
+        assertTrue(shardsResponse.valueBody().size() > 0);
+
+        ShardsRecord shardsRecord = shardsResponse.valueBody().get(0);
+
+        assertNotNull(shardsRecord.index());
+        assertNotNull(shardsRecord.shard());
+        assertNotNull(shardsRecord.prirep());
+        assertNotNull(shardsRecord.state());
+        assertNotNull(shardsRecord.store());
+        assertNull(shardsRecord.node());
+        assertNull(shardsRecord.ip());
+
+        List<String> indices = shardsResponse.valueBody().stream().map(shardRecord -> shardRecord.index()).collect(Collectors.toList());
+        assertTrue(indices.equals(indices.stream().sorted().collect(Collectors.toList())));
+    }
+
+    public void testCatAllocation() throws Exception {
+        createIndex("test-cat-allocations-index");
+
+        AllocationResponse allocationResponse = javaClient().cat().allocation(r ->
+                r.headers("ip,node,shards,disk.indices,disk.used,disk.avail,disk.total,disk.percent").sort("ip"));
+        assertNotNull(allocationResponse.valueBody());
+        assertTrue(allocationResponse.valueBody().size() > 0);
+
+        AllocationRecord allocationRecord = allocationResponse.valueBody().stream().filter(allocation ->
+                !allocation.node().equals("UNASSIGNED")).findAny().get();
+        assertNotNull(allocationRecord.ip());
+        assertNotNull(allocationRecord.node());
+        assertNotNull(allocationRecord.shards());
+        assertNotNull(allocationRecord.diskIndices());
+        assertNotNull(allocationRecord.diskUsed());
+        assertNotNull(allocationRecord.diskAvail());
+        assertNotNull(allocationRecord.diskTotal());
+        assertNotNull(allocationRecord.diskPercent());
+        assertNull(allocationRecord.host());
+    }
+
+    public void testCatPlugins() throws Exception {
+        assumeTrue("There are no plugins installed when HTTP is used", isHttps());
+        PluginsResponse pluginsResponse = javaClient().cat().plugins();
+        assertNotNull(pluginsResponse.valueBody());
+        assertTrue(pluginsResponse.valueBody().size() > 0);
+    }
+
+    public void testCatSegments() throws Exception {
+        createIndex("cat-segments-test-index");
```

Comment:
```suggestion
        createIndex("cat-segments-test-index");
        
        final IndexResponse index = javaClient().index(b -> b
            .index("test-cat-segments-index")
            .id("id")
            .refresh(Refresh.True)
            .document(Map.of()));
        assertTrue(index.result() == Result.Created);
```

Reply:
We need to add at lest one document to be 100% there would be at least one segment